### PR TITLE
Add group selector & routing

### PR DESCRIPTION
**Pull request by vtan**
- Adds a new page/screen that lists all groups and links to the group pages.
- Adds routing, so appname.com is the group list and appname.com/groupname shows the group with that name.
- To support this, the `name` and `urlName` fields are added to `Group`, and a new endpoint is added for getting all groups.

### CODE REVIEW

File: src/main/typescript/components/App.tsx [hunk 1]
```diff
@@ -1,47 +1,103 @@
...
+    </Container>
+  </BrowserRouter>
+}
+
+interface GroupPageProps {
+  dispatch: AppReducer.Dispatch,
+  state: AppReducer.State
+}
+
+function GroupPage({ dispatch, state }: GroupPageProps) {
```
Comment (vtan) [inline src/main/typescript/components/App.tsx:41]: This component could/should get its own file, I've kept it here for now to keep the diff ~small.

Comment (ejulcsi) [reply src/main/typescript/components/App.tsx:41]: it is pretty messy indeed, but ok, understood why

File: src/main/typescript/components/App.tsx [hunk 2]
```diff
@@ -1,47 +1,103 @@
...
+  const { groups, openGroup, showAddForm } = state
+
+  const { groupUrlName } = useParams()
+
+  React.useEffect(
+    function closeGroupOnUnmount() {
```
Comment (ejulcsi) [inline src/main/typescript/components/App.tsx:47]: the callback of the useEffect can be more simple, like 

`
 React.useEffect(() => {
        dispatch({ type: "groupClosed" })
        AppReducer.fetchGroups(dispatch)
    },
    []
  )`

**Review by ejulcsi — APPROVED** [review]

File: src/main/typescript/components/App.tsx (hunk 2 above)
Comment (vtan) [reply src/main/typescript/components/App.tsx:47]: I used a named function to make the intent clearer, but it arguably doesn't do much for a 2-line function.

**Pull request merged**